stdlib/test/collections/test_list.mojo

```diff
@@ -759,6 +759,98 @@ def test_list_contains():
     # assert_equal(List(0,1) in y,False)
 
 
+from sys.intrinsics import _type_is_eq
+
+
+def test_list_eq():
+    var a = List[Int](1, 2, 3)
+    var b = List[Int](1, 2, 3)
+    var c = List[Int](0, 1, 2)
+    var d = List[Int](0, 1, 2, 3)
+    assert_true(a.__eq__(b))
+    assert_false(a.__eq__(c))
+    assert_false(a.__eq__(d))
+    assert_true(a == b)
+    assert_false(a == c)
+    assert_false(a == d)
+
+    var e = List[StringLiteral]("Mojo", "Is", "Awesome")
```

Comment:
**Suggestion** Want to use `alias` instead here for the variables and we can make sure `eq` and `ne` work in comp-time?